lib/internal/dns/utils.js

```diff
@@ -23,10 +24,17 @@ const {
   ERR_INVALID_OPT_VALUE
 } = errors.codes;
 
+function validateTimeout(options) {
+  const { timeout = -1 } = { ...options };
+  validateInt32(timeout, 'options.timeout', -1, 2 ** 31 - 1);
```

Comment:
```suggestion
  validateInt32(timeout, 'options.timeout', -1, 1000);
```

This is the current maximum allowed and coercing values to something else than what the user passed through is somewhat surprising. Throwing provides a better feedback to the user for such cases.

Reply:
I think you're conflating the timeout and the poll interval. The former is when c-ares gives up, the latter the interval at which c-ares checks for new events. (There's room for optimizing latency with the latter but that aside.)

The clamping that takes place in C++ land is for the poll interval, not the timeout. That's also why the default is _not_ 1000, it's about 75x that. :-)

Reply:
Thanks for the clarification!